images/kubectl-build-deploy-dind/build-deploy-docker-compose.sh

```diff
@@ -933,16 +941,71 @@ if [[ "${CAPABILITIES[@]}" =~ "backup.appuio.ch/v1alpha1/Schedule" ]]; then
   fi
   TEMPLATE_PARAMETERS+=(-p BAAS_BUCKET_NAME="${BAAS_BUCKET_NAME}")
 
-  # Run Backups every day at 2200-0200
-  BACKUP_SCHEDULE=$( /kubectl-build-deploy/scripts/convert-crontab.sh "${NAMESPACE}" "M H(22-2) * * *")
+  # Pull in .lagoon.yml variables
+  LAGOON_YAML_KEYS=($(cat .lagoon.yml | shyaml keys))
+  if containsElement "production-backup-retention" "${LAGOON_YAML_KEYS[@]}"; then
+    LAGOON_YAML_BACKUP_RETENTION_KEYS=($(cat .lagoon.yml | shyaml keys production-backup-retention))
+
+    if containsElement "monthly" "${LAGOON_BACKUP_RETENTION_YAML_KEYS[@]}"; then
+      PRODUCTION_MONTHLY_BACKUP_RETENTION=$(cat .lagoon.yml | shyaml keys production-backup-retention.monthly -q)
+    else
+      PRODUCTION_MONTHLY_BACKUP_RETENTION=""
+    fi
+
+    if containsElement "weekly" "${LAGOON_BACKUP_RETENTION_YAML_KEYS[@]}"; then
+      PRODUCTION_WEEKLY_BACKUP_RETENTION=$(cat .lagoon.yml | shyaml keys production-backup-retention.weekly -q)
+    else
+      PRODUCTION_WEEKLY_BACKUP_RETENTION=""
+    fi
+
+    if containsElement "daily" "${LAGOON_BACKUP_RETENTION_YAML_KEYS[@]}"; then
+      PRODUCTION_DAILY_BACKUP_RETENTION=$(cat .lagoon.yml | shyaml keys production-backup-retention.daily -q)
+    else
+      PRODUCTION_DAILY_BACKUP_RETENTION=""
+    fi
+  else
+    PRODUCTION_MONTHLY_BACKUP_RETENTION=""
+    PRODUCTION_WEEKLY_BACKUP_RETENTION=""
+    PRODUCTION_DAILY_BACKUP_RETENTION=""
+  fi
+
+  # Pull in environment type (development/production)
+  LAGOON_ENVIRONMENT_TYPE=${ENVIRONMENT_TYPE}
+  TEMPLATE_PARAMETERS+=(-p LAGOON_ENVIRONMENT_TYPE="${ENVIRONMENT_TYPE}")
+
+  # Set template parameters for retention values (prefer .lagoon.yml values over supplied defaults after ensuring they are valid integers)
+  if [ ! -z $PRODUCTION_MONTHLY_BACKUP_RETENTION ] && [ "$PRODUCTION_MONTHLY_BACKUP_RETENTION" -eq "$PRODUCTION_MONTHLY_BACKUP_RETENTION" ] && [ $LAGOON_ENVIRONMENT_TYPE = 'production'] 2>/dev/null; then
```

Comment:
why do we do this here?
```
[ "$PRODUCTION_MONTHLY_BACKUP_RETENTION" -eq "$PRODUCTION_MONTHLY_BACKUP_RETENTION" ]
```
we compare the same to the same?

also the `2>/dev/null` probably is not needed? as you don't run any commands that have output?

Reply:
Comparing the value to itself via the `-eq` operator is how we ensure this value is an integer, as the `-eq` operator will attempt to parse a string into an int and fail if such conversion is not possible. The output redirection just squashes the output from those operations if the conversion fails.

Reply:
ah, ok, but let's document this here then that the `-eq` is used to make sure it's an integer, as currently it looks like a bug :)

and please don't redirect errors into `/dev/null` unless you have a specific handling for it with a custom error message